src/gui/providers/gdal/qgsgdalsourceselect.cpp

```diff
@@ -189,13 +234,157 @@ void QgsGdalSourceSelect::addButtonClicked()
       uri = protocolURI->text();
     }
 
-    QString dataSource = createProtocolURI( cmbProtocolTypes->currentText(),
-                                            uri,
-                                            mAuthSettingsProtocol->configId(),
-                                            mAuthSettingsProtocol->username(),
-                                            mAuthSettingsProtocol->password() );
-    emit addRasterLayer( dataSource, dataSource, QStringLiteral( "gdal" ) );
+    QVariantMap parts;
+    if ( !openOptions.isEmpty() )
+      parts.insert( QStringLiteral( "openOptions" ), openOptions );
+    parts.insert( QStringLiteral( "path" ),
+                  createProtocolURI( cmbProtocolTypes->currentText(),
+                                     uri,
+                                     mAuthSettingsProtocol->configId(),
+                                     mAuthSettingsProtocol->username(),
+                                     mAuthSettingsProtocol->password() ) );
+    mDataSources << QgsProviderRegistry::instance()->encodeUri( QStringLiteral( "gdal" ), parts );
   }
 }
 
+void QgsGdalSourceSelect::clearOpenOptions()
+{
+  mOpenOptionsWidgets.clear();
+  mOpenOptionsGroupBox->setVisible( false );
+  mOpenOptionsLabel->clear();
+  while ( mOpenOptionsLayout->count() )
+  {
+    QLayoutItem *item = mOpenOptionsLayout->takeAt( 0 );
+    delete item->widget();
+    delete item;
+  }
+}
+
+void QgsGdalSourceSelect::fillOpenOptions()
+{
+  clearOpenOptions();
+  computeDataSources();
+  if ( mDataSources.isEmpty() )
+    return;
+
+  GDALDriverH hDriver;
+  if ( STARTS_WITH_CI( mDataSources[0].toUtf8().toStdString().c_str(), "PG:" ) )
```

Comment:
This check should be removed. For the raster side, it is expected that the PostgisRaster driver triggers for a GDAL dataset name starting with PG: